Kernel/Bus/USB/UHCI/UHCIController.cpp

```diff
@@ -510,6 +519,41 @@ ErrorOr<size_t> UHCIController::submit_bulk_transfer(Transfer& transfer)
     return transfer_size;
 }
 
+ErrorOr<void> UHCIController::submit_async_interrupt_transfer(NonnullLockRefPtr<Transfer> transfer, u16 ms_interval)
+{
+    dbgln_if(UHCI_DEBUG, "UHCI: Received interrupt transfer for address {}. Root Hub is at address {}.", transfer->pipe().device_address(), m_root_hub->device_address());
+
+    if (ms_interval == 0) {
+        return EINVAL;
+    }
+
+    SpinlockLocker locker(m_async_lock);
+
+    auto transfer_queue = TRY(create_transfer_queue(*transfer));
+
+    u8 idx = 0;
+    while ((1 << idx) < ms_interval && idx < NUMBER_OF_INTERRUPT_QHS) {
+        idx++;
+    }
```

Comment:
Is this trying to bin the transfer based on the log base 2 of the millisecond interval? It's not quite clear whether we really need a loop here but my refactoring brain isn't finding a cleaner solution using some fancy bit manip methods.

Reply:
Yeah that's pretty much what it's doing. Basically we need to round it to a power of two so we can link it in to the closest schedule queue (the power of two part is only really required for high-speed interrupt endpoints & isochronous endpoints, but it's a good enough approximation and makes things easier for full/low speed interrupt transfers anyway, I will add a comment for this). We could use log2 here but it would involve float casts & arithmetic which is kind of awkward.